storage/store.go

```diff
@@ -493,20 +494,26 @@ func (s *Store) Start(stopper *util.Stopper) error {
 	return nil
 }
 
+// WaitForInit waits for any asynchronous processes begun in Start()
+// to complete their initialization. In particular, this includes
+// gossiping. In some cases this may block until the range GC queue
+// has completed its scan. Only for testing.
+func (s *Store) WaitForInit() {
+	s.initComplete.Wait()
+}
+
 // startGossip runs an infinite loop in a goroutine which regularly checks
 // whether the store has a first range or config replica and asks those ranges
 // to gossip accordingly.
 func (s *Store) startGossip() error {
-	// Go through one iteration synchronously before returning. This makes sure
-	// that everything is gossiped when the store finishes starting.
-	if err := s.maybeGossipConfigs(); err != nil {
-		return err
-	}
-	if err := s.maybeGossipFirstRange(); err != nil {
-		return err
-	}
+	s.initComplete.Add(2)
```

Comment:
This is racey. If the waiter gets to `WaitForInit` first, then it will return immediately. Why don't you add the `initComplete.Add(2)` invocation to `NewStore` with a comment explaining it's initialized to 2 because both gossip for first range and gossip for configs must have a chance to run before the store is considered fully initialized. Also mention that it's for unittesting.